### PR TITLE
[v6] Add react-bootstrap theme to replace bootstrap-4 (#3638)

### LINKED ISSUE #3638: Dropping bootstrap-4 support by renaming theme to react-bootstrap, using version 2
### Prerequisites

- [X] I have read the [documentation](https://rjsf-team.github.io/react-jsonschema-form/docs)

### What theme are you using?

bootstrap-4

### Is your feature request related to a problem? Please describe.

Given that the `bootstrap-4` theme is based on the `react-bootstrap` package and we are 1 major release behind we need to deal with both of those issues at the same time.

### Describe the solution you'd like

We are planning to rename the theme in the next major release. At the same time, we will bump it to the react-bootstrap 2 version. This will be a breaking change

### Describe alternatives you've considered

Provide a `bootstrap-5` theme instead. We ruled this out because:
1. It is 99% the same code
2. Uses the same package name `react-bootstrap`
3. Is inaccurately named
4. Bootstrap "4" is no longer supported

### PR DESCRIPTION
**Pull request by nickgros**
Fixes #3638 
 - Add react-bootstrap package and use react-bootstrap v2, corresponding to Bootstrap 5
 - CheckboxesWidget: Remove deprecated prop `custom`
 - IconButton: Remove deprecated `block` prop
 - RangeWidget: Use `FormRange` component
 - SelectWidget: Use new FormSelect component, remove `bsPrefix` prop to achieve correct styling

- [X] **I'm updating documentation**
  - [ ] I've [checked the rendering](https://rjsf-team.github.io/react-jsonschema-form/docs/contributing) of the Markdown text I've added
- [X] **I'm adding or updating code**
  - [X] I've added and/or updated tests. I've run `npm run test:update` to update snapshots, if needed.
  - [X] I've updated [docs](https://rjsf-team.github.io/react-jsonschema-form/docs) if needed
  - [X] I've updated the [changelog](https://github.com/rjsf-team/react-jsonschema-form/blob/main/CHANGELOG.md) with a description of the PR
- [X] **I'm adding a new feature**
  - [X] I've updated the playground with an example use of the feature

### CODE REVIEW

File: packages/playground/package.json [hunk 1]
```diff
@@ -41,13 +41,13 @@
...
     "@mui/material": "^5.14.8",
     "@rjsf/antd": "^5.14.2",
-    "@rjsf/bootstrap-4": "^5.14.2",
     "@rjsf/chakra-ui": "^5.14.2",
     "@rjsf/core": "^5.14.2",
     "@rjsf/fluent-ui": "^5.14.2",
     "@rjsf/fluentui-rc": "^5.14.2",
     "@rjsf/material-ui": "^5.14.2",
     "@rjsf/mui": "^5.14.2",
+    "@rjsf/react-bootstrap": "file:../react-bootstrap",
```
Comment (nickgros) [inline packages/playground/package.json:50]: This lets me build the playground without releasing the theme to NPM. We can update this to the version number once we release a v6 beta.

**Comment by nickgros** [comment]
Converting to draft until we get CI working for rjsf-v6 branch

**Comment by heath-freenome** [comment]
@nickgros I guess the first thing we need to do is create the long-lived branch

File: CHANGELOG.md [hunk 2]
```diff
@@ -15,6 +15,21 @@ it according to semantic versioning. For example, if your PR adds a breaking cha
 should change the heading of the (upcoming) version to include a major version bump.
 
 -->
+# 6.0.0
```
Comment (heath-freenome) [inline CHANGELOG.md:18]:
```suggestion
# 6.0.0-beta.1
```

File: packages/react-bootstrap/README.md [hunk 3]
```diff
@@ -39,12 +39,12 @@
...
 ## About The Project
 
-Exports `bootstrap-4` theme, fields and widgets for `react-jsonschema-form`.
+Exports `react-bootstrap` theme, fields and widgets for `react-jsonschema-form`.
 
 ### Built With
 
 - [react-jsonschema-form](https://github.com/mozilla-services/react-jsonschema-form/)
-- [Bootstrap-4](https://getbootstrap.com/docs/4.0/)
+- [Bootstrap-5](https://getbootstrap.com/docs/4.0/)
```
Comment (heath-freenome) [inline packages/react-bootstrap/README.md:47]: Update this to 5

File: packages/react-bootstrap/package.json [hunk 4]
```diff
@@ -1,10 +1,10 @@
 {
-  "name": "@rjsf/bootstrap-4",
+  "name": "@rjsf/react-bootstrap",
   "version": "5.15.0",
   "main": "dist/index.js",
   "module": "lib/index.js",
   "typings": "lib/index.d.ts",
-  "description": "Bootstrap 4 theme, fields and widgets for react-jsonschema-form",
+  "description": "Bootstrap 5 theme, fields and widgets for react-jsonschema-form, powered by react-bootstrap",
```
Comment (heath-freenome) [inline packages/react-bootstrap/package.json:7]: Consider calling this `React Bootstrap` theme

File: packages/react-bootstrap/package.json [hunk 5]
```diff
@@ -74,16 +74,16 @@
...
     "jsonschema",
     "json-schema",
     "json",
     "schema",
     "form",
     "react",
-    "bootstrap-4",
-    "rjsf-bootstrap-4"
+    "bootstrap-5",
+    "rjsf-bootstrap-5"
```
Comment (heath-freenome) [inline packages/react-bootstrap/package.json:86]: add `react-bootstrap`, `rjsf-react-bootstrap`

File: CHANGELOG.md [hunk 6]
```diff
@@ -15,6 +15,21 @@ it according to semantic versioning. For example, if your PR adds a breaking cha
 should change the heading of the (upcoming) version to include a major version bump.
 
 -->
+# 6.0.0-beta.1
+
+## @rjsf/bootstrap-4
+
+- Package has been replaced with `@rjsf/react-bootstrap`. `react-boostrap` v1 / Bootstrap 4 are no longer supported in RJSF v6.
```
Comment (heath-freenome) [inline CHANGELOG.md:22]: Also, add this to the migration guide with breaking changes?

**Review by heath-freenome — APPROVED** [review]

**Pull request merged**